src/UIKit/UITraitCollection.cs

```diff
@@ -14,7 +14,7 @@
 namespace UIKit {
 #if !WATCH
 	public partial class UITraitCollection {
-		public UITraitCollection FromPreferredContentSizeCategory (UIContentSizeCategory category)
+		public static UITraitCollection FromPreferredContentSizeCategory (UIContentSizeCategory category)
```

Comment:
As @spouliot mentioned, it is changing the API. Since we are moving to static.. I wonder if we can be smart, leave the instance  one and call the static one from that. Expose both and remove the instance one on XAMCORE_4.

Reply:
Something like - 
```
#if XAMCORE_4_0
		[Obsolete ("Please use the static method instead.")]
#endif
		public UITraitCollection FromPreferredContentSizeCategory (UIContentSizeCategory category)
		{
			return FromPreferredContentSizeCategory (category.GetConstant ());
		}

#if XAMCORE_4_0
		public static UITraitCollection FromPreferredContentSizeCategory (UIContentSizeCategory category)
		{
			return FromPreferredContentSizeCategory (category.GetConstant ());
		}
#endif
```
should work?

Reply:
Also, the instance one is already calling an internal static method

Reply:
Close, we could name the static method create since it is a factory method, and do:
```csharp
#if !XAMCORE_4_0
[Obsolete ("Please use the static method 'StaticFromPreferredContentSizeCategory' instead.")]
public UITraitCollection FromPreferredContentSizeCategory (UIContentSizeCategory category) => Create (category);
#endif

public static UITraitCollection Create (UIContentSizeCategory category)
{
  return FromPreferredContentSizeCategory (category.GetConstant ());
}
```

@spouliot  thoughts?

Reply:
close but not quite :)

We want to avoid breaking changes _before_ `XAMCORE_4_0` so we add an `[Obsolete]` until it's defined. Once it's define then we can remove the old code (and attribute) from the build. E.g.

```csharp
#if !XAMCORE_4_0
[Obsolete ("Please use the static method 'StaticFromPreferredContentSizeCategory' instead.")]
public UITraitCollection FromPreferredContentSizeCategory (UIContentSizeCategory category)
{
  return FromPreferredContentSizeCategory (category.GetConstant ());
}
#endif

public static UITraitCollection StaticFromPreferredContentSizeCategory (UIContentSizeCategory category)
{
  return FromPreferredContentSizeCategory (category.GetConstant ());
}
```

Reply:
Actually, won't this cause an error considering we have a static and an instance method of the same signature?